packages/web-components/src/components/va-button/va-button.tsx

```diff
@@ -76,17 +76,20 @@ export class VaButton {
   })
   componentLibraryAnalytics: EventEmitter;
 
-  private handleClick = (e: MouseEvent): void => {
-    // TODO: remove console.log
-    console.log(e);
-    if (this.disableAnalytics) return;
-    this.componentLibraryAnalytics.emit({
-      componentName: 'va-button',
-      action: 'click',
-      details: {
-        // TODO: add analytics event details
-      },
-    });
+  private handleClick = (): void => {
+    if (!this.disableAnalytics) {
+      const detail = {
+        componentName: 'va-button',
+        action: 'click',
+        details: {
+          event: 'cta-button-click',
+          'button-type': this.secondary ? 'secondary' : 'primary',
+          'button-click-label': this.getButtonText(),
```

Comment:
I think it would be good if we standardized around a set vocabulary for common things. We're using `button-click-label` here to represent the "label" or text of the button, and in the radio component we're just using the simpler `label` to match the prop name:

https://github.com/department-of-veterans-affairs/component-library/blob/bb1bcb44d2eb724fbbe70da96f8e3bb06f7eb551/packages/web-components/src/components/va-radio/va-radio.tsx#L161-L165

To make things worse, the `label` prop on the button is meant to be a pass-through for an `aria-label`:

https://github.com/department-of-veterans-affairs/component-library/blob/bb1bcb44d2eb724fbbe70da96f8e3bb06f7eb551/packages/web-components/src/components/va-button/va-button.tsx#L130-L133

So I'm not saying "this needs to have the name 'label'", but I'm hoping we can avoid getting into a situation where our analytics events are using many different names to represent the same thing, and I'm not sure that the `button-` prefix adds anything here since we already know it's a button from the `componentName`.

Reply:
I dug into this a little more and it looks like we should probably set the key value at this point as just "label" because [in the vets-website script](https://github.com/department-of-veterans-affairs/vets-website/blob/77e40f5431f228ffba7a159dd377f5a09d3e4d9b/src/platform/site-wide/component-library-analytics-setup.js#L350), it's transformed to `${action.prefix}-${key}`. For example, in the va-radio example, the actual datalayer key that will be sent for label after it's transformed is `button-label`. 

Note: I think the use of the value `action` is kind of a poor name in this case because there is also an `action` within the component event details. "action" in this case is just the found component info object in vets-website like below:

```
  'va-button': [
    {
      action: 'click',
      event: 'cta-button-click',
      prefix: 'button',
    },
  ],
```